server/backend/server.js

```diff
@@ -52,19 +54,15 @@ var decrypt = (req) => {
   1. Queries `segments` collection to find the road the coordinate lies on.
   2. If matched, queries `roads` collection to determine road name.
 
-  Returns a Response object.
+  Returns a Promise wrapped Response object.
 */
-var findRoad = (parsed, db, callback) => {
-  const resp = newResponse();
-
+const findRoad = ({lat, lng}, db) => {
```

Comment:
`findRoad` should only be returning a Promise wrapped Response object. I'll comment on the lines with explanations.